### PR TITLE
Allow specifying firmware path with firmware.unpack

### PR DESCRIPTION
**Pull request by jjcarstens**
`mix firmware.unpack` would only look for a firmware that was currently built
for a project. Sometimes you may have multiple firmware files to inspect that
were built elsewhere and this would be a convenient tool to inspect them.

This adds `--fw path/to/.fw` support for those cases. It also changes the default
outpath to be the firmware file name with `.unpacked` appended. (i.e. `hello_nerves.fw -> hello_nerves.unpacked`)

### CODE REVIEW

File: lib/mix/tasks/firmware.unpack.ex [hunk 1]
```diff
@@ -24,54 +29,37 @@ defmodule Mix.Tasks.Firmware.Unpack do
   # Unpack it
   mix firmware.unpack firmware_contents
 
+  # Unpack specified fw file
+  mix firmware.unpack --fw hello_nerves.fw
+
   # Inspect it
   ls firmware_contents
```
Comment (fhunleth) [inline lib/mix/tasks/firmware.unpack.ex:28]: This line isn't correct any more.

**Review by fhunleth — COMMENTED** [review]
It feels really odd to me having the input being an option and the output directory being a positional arg now. Since changing the default output name is an API breaking change anyway, how about making the output directory an option too?

**Comment by jjcarstens** [comment]
@fhunleth Changed to support `--output` option as well

**Review by fhunleth — APPROVED** [review]
Looks good to me! I like you changed the default output directory to something that makes more sense.

**Review by mobileoverlord — APPROVED** [review]

**Pull request merged**